bigc/api_client.py

```diff
@@ -17,16 +17,22 @@ def __init__(self, store_hash: str, access_token: str, timeout: float | None = N
         self.timeout = timeout
 
     def request(
-        self,
-        method: str,
-        path: str,
-        *,
-        data: Any = None,
-        params: dict[str, Any] | None = None,
-        headers: dict[str, str] | None = None,
-        timeout: float | None = None,
+            self,
+            method: str,
+            path: str,
+            *,
+            data: Any = None,
+            params: dict[str, Any] | None = None,
+            headers: dict[str, str] | None = None,
+            timeout: float | None = None,
+            retry_attempts: int | None = None,
     ) -> Any:
-        """Make a request to the BigCommerce API (uses Requests internally)"""
+        """Make a request to the BigCommerce API (uses Requests internally) with support for retries"""
```

Comment:
Maybe just get rid of this docstring since we are trying to encapsulate Requests now

Reply:
https://github.com/MedShift/bigc/pull/55/commits/034cc973f1cd9bbbd13273ef7fc9e7f1cdf52fa4